### ISSUE TITLE
Disallow covariant override of instance or class variables

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1348]
```python
class A:
    x: float

class B(A):
    x: int  # Should generate error

def f(a: A) -> None:
    a.x = 1.1

b = B()
f(b)
b.x  # float at runtime, though declared as int

```

**Comment by erictraut** [comment]
Upon further thought, this should probably be allowed. Subclasses that narrow the type of a member need to know that the base class doesn't update the member with an incompatible type.
